### PR TITLE
Replace Console.WriteLine with Logging in generic host

### PR DESCRIPTION
**Pull request by Tratcher**
https://github.com/aspnet/Hosting/issues/1584
@glennc

### CODE REVIEW

File: src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs [hunk 1]
```diff
@@ -28,15 +34,17 @@ public ConsoleLifetime(IOptions<ConsoleLifetimeOptions> options, IHostingEnviron
 
         private IApplicationLifetime ApplicationLifetime { get; }
 
+        private  ILogger Logger { get; }
```
Comment (pakrym) [inline src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs]: nit space

File: src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs [hunk 2]
```diff
@@ -16,10 +18,14 @@ public class ConsoleLifetime : IHostLifetime, IDisposable
...
 
         public ConsoleLifetime(IOptions<ConsoleLifetimeOptions> options, IHostingEnvironment environment, IApplicationLifetime applicationLifetime)
+            : this(options, environment, applicationLifetime, NullLoggerFactory.Instance) { }
+
+        public ConsoleLifetime(IOptions<ConsoleLifetimeOptions> options, IHostingEnvironment environment, IApplicationLifetime applicationLifetime, ILoggerFactory loggerFactory)
         {
             Options = options?.Value ?? throw new ArgumentNullException(nameof(options));
             Environment = environment ?? throw new ArgumentNullException(nameof(environment));
             ApplicationLifetime = applicationLifetime ?? throw new ArgumentNullException(nameof(applicationLifetime));
+            Logger = loggerFactory.CreateLogger("Microsoft.Extensions.Hosting.Host");
```
Comment (pakrym) [inline src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs]: `CreateLogger<Host>`?

Comment (Tratcher) [reply src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs]: Microsoft.Extensions.Hosting.Host doesn't actually exist, it's Microsoft.Extensions.Hosting.Internal.Host. I didn't want Internal in the log scope.

Comment (pakrym) [reply src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs]: What about https://github.com/aspnet/Extensions/blob/master/src/Hosting/Hosting/src/Internal/Host.cs#L23 then?

Comment (Tratcher) [reply src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs]: Yeah, we should re-think that. We've changed that kind of thing elsewhere. @glennc

Comment (Tratcher) [reply src/Hosting/Hosting/src/Internal/ConsoleLifetime.cs]: Offline note: making the pubternal types properly internal and flattening the namespace would indirectly deal with this. We've taken that approach in other projects like MVC.

**Pull request merged**